lib/iris/tests/integration/netcdf/test_ugrid_load.py

```diff
@@ -45,167 +42,177 @@ def ugrid_load(uris, constraints=None, callback=None):
-            temp_file_dir=self.temp_dir,
+            temp_file_dir=temp_dir,
             dataset_name=testfile_name,
             dataset_type=template_name,
             template_subs=template_subs,
         )
         return str(filepath)  # N.B. Path object not usable in iris.load
 
-    def test_mesh_bad_topology_dimension(self):
+    def test_mesh_bad_topology_dimension(self, tmp_path):
         # Check that the load generates a suitable warning.
         warn_regex = r"topology_dimension.* ignoring"
         with pytest.warns(IrisCfWarning, match=warn_regex):
             template = "minimal_bad_topology_dim"
             dim_line = "mesh_var:topology_dimension = 1 ;"  # which is wrong !
             cube = self.create_synthetic_test_cube(
-                template=template, subs=dict(TOPOLOGY_DIM_DEFINITION=dim_line)
+                temp_file_dir=tmp_path,
+                template=template,
+                subs=dict(TOPOLOGY_DIM_DEFINITION=dim_line),
             )
 
         # Check that the result has topology-dimension of 2 (not 1).
-        self.assertEqual(cube.mesh.topology_dimension, 2)
+        assert cube.mesh.topology_dimension == 2
 
-    def test_mesh_no_topology_dimension(self):
+    def test_mesh_no_topology_dimension(self, tmp_path):
         # Check that the load generates a suitable warning.
         warn_regex = r"MeshXY variable.* has no 'topology_dimension'"
         with pytest.warns(IrisCfWarning, match=warn_regex):
             template = "minimal_bad_topology_dim"
             dim_line = ""  # don't create ANY topology_dimension property
             cube = self.create_synthetic_test_cube(
-                template=template, subs=dict(TOPOLOGY_DIM_DEFINITION=dim_line)
+                temp_file_dir=tmp_path,
+                template=template,
+                subs=dict(TOPOLOGY_DIM_DEFINITION=dim_line),
             )
 
         # Check that the result has the correct topology-dimension value.
-        self.assertEqual(cube.mesh.topology_dimension, 2)
+        assert cube.mesh.topology_dimension == 2
 
-    def test_mesh_bad_cf_role(self):
+    def test_mesh_bad_cf_role(self, tmp_path):
         # Check that the load generates a suitable warning.
         warn_regex = r"inappropriate cf_role"
         with pytest.warns(IrisCfWarning, match=warn_regex):
             template = "minimal_bad_mesh_cf_role"
             dim_line = 'mesh_var:cf_role = "foo" ;'
             _ = self.create_synthetic_test_cube(
-                template=template, subs=dict(CF_ROLE_DEFINITION=dim_line)
+                temp_file_dir=tmp_path,
+                template=template,
+                subs=dict(CF_ROLE_DEFINITION=dim_line),
             )
 
-    def test_mesh_no_cf_role(self):
+    def test_mesh_no_cf_role(self, tmp_path):
         # Check that the load generates a suitable warning.
         warn_regex = r"no cf_role attribute"
         with pytest.warns(IrisCfWarning, match=warn_regex):
```

Comment:
`PT031`: should only be one line in the indented block.

Reply:
https://github.com/SciTools/iris/pull/6646/commits/db72d6a96cb0a15e971547f1bb448d7431ef8e64